### PR TITLE
fix(security): deny cross-profile state access by default

### LINKED ISSUE #1784: Security: add safe default for cross-profile state access
## Description
Implement an LLM-friendly, independently shippable backlog item for **add safe default for cross-profile state access**. This issue should stay narrow enough for one agent/PR while producing durable behavior, tests, and documentation where appropriate.

## Affected locations
- Approval/governor/security policy surfaces
- Runtime state, logs, prompts, and delivery integrations
- Agent tool wrappers and web/control endpoints

## Recommendation
Add the smallest safe control that addresses this risk, prefer deny-by-default behavior for unsafe paths, document operator override semantics, and include regression tests for both allowed and denied flows.

## Goals
Deliver add safe default for cross-profile state access as a discrete capability or hardening change.
Make the behavior deterministic enough for autonomous workers to verify.
Keep the implementation small enough for a one-issue worker and one PR.
Capture operator/user-facing behavior in docs or help text when applicable.

## Expected outcomes
add safe default for cross-profile state access is available, documented, and test-backed.
Future PM/worker handoffs can point at this issue without needing extra interpretation.
Failure modes are explicit and produce actionable messages instead of silent drift.
The change does not broaden into adjacent backlog work or unrelated refactors.

## Suggested implementation slices
Inventory the current code/docs paths that own this behavior.
Add or update the narrow model/config/schema needed for the feature or guardrail.
Implement the smallest runtime/docs change that satisfies the acceptance criteria.
Add focused fixtures/tests for success, failure, and edge cases.
Update docs/help output and handoff notes so later agents can use it.

## Acceptance criteria
- [ ] There is a dedicated implementation or documentation change for add safe default for cross-profile state access.
- [ ] The control defaults to the safest practical behavior and requires explicit override for risky cases.
- [ ] At least one automated test, fixture, or deterministic verifier covers the intended behavior.
- [ ] Sensitive payloads/secrets are not printed in logs, test output, issue comments, or telemetry.
- [ ] At least one negative/edge case is covered so the failure mode is explicit.
- [ ] Docs/help text or inline operator guidance explains how to use or interpret the behavior.
- [ ] The PR description links this issue and summarizes verification evidence.
- [ ] No unrelated production behavior is changed outside the stated scope.

## LLM handoff notes
- Treat this as a one-issue/one-PR unit.
- Start by searching for existing implementations and tests before adding new surfaces.
- Prefer structured outputs and fixtures over prose-only behavior.
- If this overlaps an existing open issue, comment with evidence and stop for PM deduplication instead of creating an unscoped implementation.
- Do not perform broad refactors or change unrelated worker/PM behavior while implementing this ticket.

## Source
Created by the 2026-07-11 fourth strategic backlog expansion run requesting 100 issues. Topic: **Security**.

### PR DESCRIPTION
**Pull request by djm204**
## Summary
- fail closed when orchestrator stateDir points at a sibling Hermes profile unless explicitly allowed
- add config default/docs/examples for allowCrossProfileStateAccess
- add regression coverage for denied and explicitly allowed cross-profile state paths

## Test Plan
- npm test --workspace @franken/orchestrator -- --run tests/unit/config/orchestrator-config.test.ts
- npm test --workspace @franken/orchestrator
- npm run typecheck --workspace @franken/orchestrator
- npm run lint --workspace @franken/orchestrator
- npm run build --workspace @franken/orchestrator

Closes #1784

### CODE REVIEW

**Comment by djm204** [comment]
@codex review

File: packages/franken-orchestrator/src/config/orchestrator-config.ts [hunk 1]
```diff
@@ -152,6 +171,22 @@ function createOrchestratorConfigSchema(options: OrchestratorConfigParseOptions
   ).superRefine((config, ctx) => {
     validateNetworkConfig(config, ctx);
 
+    const targetProfile = config.stateDir ? hermesProfileFromPath(config.stateDir) : undefined;
+    const currentProfile = activeHermesProfile();
+    if (
+      targetProfile !== undefined &&
+      targetProfile !== currentProfile &&
+      !config.allowCrossProfileStateAccess
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/config/orchestrator-config.ts]: Fixed by stripping allowCrossProfileStateAccess from repository-local config before schema parsing. Only explicit operator-owned config files outside the repository can carry the opt-in now, and tests cover repo-local stripping plus operator-owned preservation.

File: packages/franken-orchestrator/src/config/orchestrator-config.ts [hunk 2]
```diff
@@ -152,6 +171,22 @@ function createOrchestratorConfigSchema(options: OrchestratorConfigParseOptions
   ).superRefine((config, ctx) => {
     validateNetworkConfig(config, ctx);
 
+    const targetProfile = config.stateDir ? hermesProfileFromPath(config.stateDir) : undefined;
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/config/orchestrator-config.ts]: Fixed by validating the effective runtime stateDir as well. Session now forwards the effective stateDir into BeastLoop, and BeastLoop performs the same cross-profile guard before constructing StateSnapshotStore.

File: packages/franken-orchestrator/src/config/orchestrator-config.ts [hunk 3]
```diff
@@ -94,6 +95,21 @@ export const ProvidersConfigSchema = createProvidersConfigSchema();
 const MIN_TOTAL_TOKEN_BUDGET = 10_000;
 const MIN_DURATION_MS_PER_CRITIQUE_ITERATION = 10_000;
 
+function hermesProfileFromPath(path: string): string | undefined {
+  const parts = resolve(path).split(sep);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/config/orchestrator-config.ts]: Fixed by resolving existing stateDir symlinks before extracting Hermes profile segments. The regression test now covers a repo-local .fbeast/state symlink pointing into a sibling Hermes profile.

**Comment by djm204** [comment]
@codex review

File: packages/franken-orchestrator/src/config/orchestrator-config.ts [hunk 4]
```diff
@@ -94,6 +96,58 @@ export const ProvidersConfigSchema = createProvidersConfigSchema();
 const MIN_TOTAL_TOKEN_BUDGET = 10_000;
 const MIN_DURATION_MS_PER_CRITIQUE_ITERATION = 10_000;
 
+function resolvedPathCandidates(path: string): string[] {
+  const lexical = resolve(path);
+  try {
+    const real = realpathSync(lexical);
+    return real === lexical ? [lexical] : [real, lexical];
+  } catch {
+    return [lexical];
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/config/orchestrator-config.ts]: Fixed by resolving the nearest existing ancestor when stateDir itself does not exist yet, then appending the missing suffix before Hermes-profile extraction. Added a regression test for a not-yet-created child under a symlinked ancestor that points into a sibling profile.

**Comment by djm204** [comment]
@codex review

File: packages/franken-orchestrator/src/cli/session.ts [hunk 5]
```diff
@@ -493,6 +493,9 @@ export class Session {
     if (this.config.maxTotalTokens !== undefined) {
       loopConfig.maxTotalTokens = this.config.maxTotalTokens;
     }
+    if (this.config.orchestratorConfig?.allowCrossProfileStateAccess !== undefined) {
+      loopConfig.allowCrossProfileStateAccess = this.config.orchestratorConfig.allowCrossProfileStateAccess;
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/cli/session.ts:501]: Fixed by scoping allowCrossProfileStateAccess to a configured stateDir from the same config source. Session no longer forwards the opt-in when it falls back to the repo default stateDir, and tests verify an opt-in without stateDir cannot approve a symlinked default.

File: packages/franken-orchestrator/src/cli/session.ts [hunk 6]
```diff
@@ -493,6 +493,9 @@ export class Session {
...
+    }
     loopConfig.stateDir = this.config.orchestratorConfig?.stateDir ?? paths.stateDir;
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/cli/session.ts]: Fixed by running the resolved-path cross-profile guard before scaffoldFrankenbeast(paths), so the default stateDir is validated before any state directory can be created or touched.

**Comment by djm204** [comment]
@codex review

File: packages/franken-orchestrator/src/cli/run.ts [hunk 7]
```diff
@@ -154,6 +154,23 @@ export function shouldShowMissingRunPlanGuidance(
     && planNeedsGuidance;
 }
 
+export function validateStateDirBeforeScaffold(
+  config: OrchestratorConfig,
+  paths: Pick<ReturnType<typeof getProjectPaths>, 'stateDir'>,
+): void {
+  const configuredStateDir = config.stateDir;
+  const stateDir = configuredStateDir ?? paths.stateDir;
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/cli/run.ts]: Fixed by deriving scaffoldPaths from the configured stateDir. main() now validates and scaffolds the effective stateDir (configured stateDir when present, otherwise the repo default), so an operator-owned configured stateDir no longer still creates/touches the repo default symlink.

**Comment by djm204** [comment]
@codex review
